apps/dashboard/test/integration/custom_css_files_test.rb

```diff
@@ -0,0 +1,21 @@
+require 'test_helper'
+
+class CustomCssFilesTest < ActionDispatch::IntegrationTest
+ test "should add css tags when custom_css_files configuration is set" do
+   stub_user_configuration({custom_css_files: ["test.css", "/custom/other.css"]})
+
+   get '/'
+
+   custom_css = css_select('link[data-ref="custom-css"]')
```

Comment:
Maybe instead of relying on the `data-ref` attribute maybe just do something like this? Did you just make up `data-ref` for this test? it may get lost in time what/why that's being used (if it is just being used for tests).

```
assert css_select.any { |link| link['href'] == 'test.css' && !link['nonce'].nil? }
assert css_select.any { |link| link['href'] == '/custom/test.css' && !link['nonce'].nil? }
```

Reply:
I did add `data-ref` just for the test. It is fairly common and good practice to add custom HTML attributes to elements to avoid having to rely on HTML/CSS positioning/structure for testing.

But I am happy to update to the above selectors. I had to update to this as the `.any` didn't work for me:
```ruby
   assert css_select("link") { |link| link['href'] == 'test.css' && !link['nonce'].nil? }
   assert css_select("link") { |link| link['href'] == '/custom/test.css' && !link['nonce'].nil? }
```

Reply:
Yea I'm just weary about extra attributes that we're bound to forget about in a year. I've often added `id`s to things just so I could query them in a test. I think the best pattern would be just adding an `id` or we could have a practice of using `data-test-attr` or similar.

That's my only thing -- what's this going to read like in a year when we've forgotten the details.